organizations/tests.py

```diff
@@ -39,3 +49,57 @@ def test_superuser_can_list_organizations(self):
         self.user.save()
         response = self.client.get(reverse("organizations-list"))
         self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+    def test_updating_organization_recreates_user_instance_cache(self):
+        from users.models import User
+        from rest_framework.test import APIClient
+        from plio.tests import get_new_access_token
+
+        superadmin_client = APIClient()
+        superadmin_user = User.objects.create(mobile="+919988776655", is_superuser=True)
+        superadmin_access_token = get_new_access_token(
+            superadmin_user, self.application
+        )
+        superadmin_client.credentials(
+            HTTP_AUTHORIZATION="Bearer " + superadmin_access_token.token
+        )
+
+        # verify cache data doesn't exist by default
+        cache_key_name = get_cache_key(self.user)
+
+        # make a get request
+        self.client.get(reverse("users-detail", kwargs={"pk": self.user.id}))
+        # verify cache data exists as we made a GET request for user details
+        self.assertEqual(len(cache.keys(cache_key_name)), 1)
+        self.assertEqual(len(cache.get(cache_key_name)["organizations"]), 0)
+
+        # associate the current user with the organization
+        OrganizationUser.objects.create(
+            organization=self.organization_2, user=self.user, role=self.org_admin_role
+        )
+
+        # check cache after updating organization user association
+        # this time the user should have an organization associated
+        self.assertEqual(len(cache.get(cache_key_name)["organizations"]), 1)
+        self.assertEqual(
+            cache.get(cache_key_name)["organizations"][0]["name"],
+            self.organization_2.name,
```

Comment:
note for self: look at the workflow diagram for invalidation to understand this better.
essentially, after the cache invalidation takes place, the request fetches the data again from the DB in order to send the response. In that case, this new data is stored in the cache.